alacritty_terminal/src/config/bindings.rs

```diff
@@ -127,57 +113,107 @@ pub fn default_key_bindings() -> Vec<KeyBinding> {
-        Key::F3, [logo: true]; Action::Esc("\x1b[1;3R".into());
-        Key::F4, [logo: true]; Action::Esc("\x1b[1;3S".into());
-        Key::F5, [logo: true]; Action::Esc("\x1b[15;3~".into());
-        Key::F6, [logo: true]; Action::Esc("\x1b[17;3~".into());
-        Key::F7, [logo: true]; Action::Esc("\x1b[18;3~".into());
-        Key::F8, [logo: true]; Action::Esc("\x1b[19;3~".into());
-        Key::F9, [logo: true]; Action::Esc("\x1b[20;3~".into());
-        Key::F10, [logo: true]; Action::Esc("\x1b[21;3~".into());
-        Key::F11, [logo: true]; Action::Esc("\x1b[23;3~".into());
-        Key::F12, [logo: true]; Action::Esc("\x1b[24;3~".into());
+        Key::F13; Action::Esc("\x1b[1;2P".into());
+        Key::F14; Action::Esc("\x1b[1;2Q".into());
+        Key::F15; Action::Esc("\x1b[1;2R".into());
+        Key::F16; Action::Esc("\x1b[1;2S".into());
+        Key::F17; Action::Esc("\x1b[15;2~".into());
+        Key::F18; Action::Esc("\x1b[17;2~".into());
+        Key::F19; Action::Esc("\x1b[18;2~".into());
+        Key::F20; Action::Esc("\x1b[19;2~".into());
+        Key::F21; Action::Esc("\x1b[20;2~".into());
+        Key::F22; Action::Esc("\x1b[21;2~".into());
+        Key::F23; Action::Esc("\x1b[23;2~".into());
+        Key::F24; Action::Esc("\x1b[24;2~".into());
         Key::NumpadEnter; Action::Esc("\n".into());
     );
 
+    //   Code     Modifiers
+    // ---------+---------------------------
+    //    2     | Shift
+    //    3     | Alt
+    //    4     | Shift + Alt
+    //    5     | Control
+    //    6     | Shift + Control
+    //    7     | Alt + Control
+    //    8     | Shift + Alt + Control
+    //    9     | Meta
+    //    10    | Meta + Shift
+    //    11    | Meta + Alt
+    //    12    | Meta + Alt + Shift
+    //    13    | Meta + Ctrl
+    //    14    | Meta + Ctrl + Shift
+    //    15    | Meta + Ctrl + Alt
+    //    16    | Meta + Ctrl + Alt + Shift
+    // ---------+---------------------------
+    //
+    // from: https://invisible-island.net/xterm/ctlseqs/ctlseqs.html#h2-PC-Style-Function-Keys
+    let modifiers = vec![
+        ModifiersState { shift: true, ..ModifiersState::default() },
+        ModifiersState { alt: true, ..ModifiersState::default() },
+        ModifiersState { shift: true, alt: true, ..ModifiersState::default() },
+        ModifiersState { ctrl: true, ..ModifiersState::default() },
+        ModifiersState { shift: true, ctrl: true, ..ModifiersState::default() },
+        ModifiersState { alt: true, ctrl: true, ..ModifiersState::default() },
+        ModifiersState { shift: true, alt: true, ctrl: true, ..ModifiersState::default() },
+    ];
+
+    let mut modifers_num = 2;
+    for mods in modifiers.iter() {
+        let mods_num = modifers_num.to_string();
+        bindings.extend(bindings!(
+            KeyBinding;
+            Key::Up,    [shift: mods.shift, alt: mods.alt, ctrl: mods.ctrl];
+            Action::Esc("\x1b[1;".to_owned()  + &mods_num + "A");
```

Comment:
```suggestion
            Action::Esc(format!("\x1b[1;{}{}", mods_num, "A");
```

I think this should be shorter and it doesn't require the `modifiers_num.to_string()`.